### DISCUSSION TITLE
[Question] It is possible to use pyjnius?

### DISCUSSION THREAD

**Discussion by 9acca9** [flet-dev/flet#4028] [Q&A]
### Question

Hi.
I will connect a bluetooth device (arduino) to the android phone, and my app will get and push messages to that device.
For what i found it seems that pyjnius is the appropiatted module for this.
It is possible to use that module in a Flet app?

Thanks

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by tanmay-bhatgare** [comment]
see it's ok to use any module in flet until it is a native python module(aren't written entirely in python). You can still use them in development environment but might cause error while packaging into `apk` coz these modules don't have wheel for android environment. If `pyjnius` is pure python package, you can pack your flet app to apk. More info on [docs](https://flet.dev/docs/publish/android/#introduction).

**Comment by rralen** [comment]
en la version ya integraron integracion con pyjnius pero al menos por mi parte no lo podido hacer funcionar para bluetooth

**Reply by Muddassir-Farooq-official** [reply]
Yes this module is supported by flet kindly read blogs section of flet website there you will get example code for Android and ios

**Comment by ndonkoHenri** [comment]
Docs: https://flet.dev/blog/tap-into-native-android-and-ios-apis-with-Pyjnius-and-pyobjus

**Comment by Muddassir-Farooq-official** [comment]
> Docs: https://flet.dev/blog/tap-into-native-android-and-ios-apis-with-Pyjnius-and-pyobjus
@ndonkoHenri  I have a question we can not test this during flet run main.py --Android???
